### ISSUE TITLE
FAILURE IN SERVER /b_getn index out of range

### ISSUE THREAD

**Issue by prko** [supercollider/supercollider#3873]
While testing the latetst version which I rendered under OS X, I experienced the error

> FAILURE IN SERVER /b_getn index out of range

after evaluating the following code:
```supercollider
{
var src= SinOsc.ar(440), amp= SinOsc.ar(440/20, 0, 2);
[src, amp, src * amp, (src * amp).clip(-1, 1)]
}.plot(20/440)
```
In 3.9.3, the code works well.
If the error occurs because 3.10 is not yet completed, Please close this issue.

**Comment by patrickdupuis** [comment]
The full message I'm getting on branch `3.10` is this. The last two lines appear a couple seconds after the failure message.
```
-> a Plotter
FAILURE IN SERVER /b_getn index out of range
WARNING: Buffer-streamToFloatArray failed!
Try increasing wait time
```

**Comment by miczac** [comment]
just tried this w/ 43e92443d, can confirm.
Is this related / connected to what this PR addresses for supernova?
https://github.com/supercollider/supercollider/pull/3848

**Comment by prko** [comment]
I don't know.
However, I rendered SC with the supernova-enabled option, but supernova does not boot and I booted with sclang.

When I evaluate
```supercollider
Server.supernova
```
sc returns

> -> Server

then evaluating
```
s.reboot
```
returns

> server 'localhost' disconnected shared memory interface
> '/quit' message sent to server 'localhost'.
> -> localhost
> Server 'localhost' exited with exit code 0.
> Booting server 'localhost' on address 127.0.0.1:57110.
> Error when parsing command line arguments:
> the argument ('2097152.0') for option '--rt-memory' is invalid
> 
> Server 'localhost' exited with exit code 1.

**Comment by snappizz** [comment]
i found that this is sufficient:

    { DC.ar(0) ! 2 }.plot(0.0101)

strangely, this bug appears to be sensitive to the plot duration. this consistently works for me:

    { DC.ar(0) ! 2 }.plot(0.0100)

conjecture: the duration must translate to an integer number of samples.

if you check the `Function:plot` source code, you'll fin immediately translates to `getToFloatArray`, and if you replace `plot` with `getToFloatArray` the bug is still visible.

...wait a minute, i could have sworn it was actually `loadToFloatArray` at some point. when did that change? git blame tells me that it was changed to `getToFloatArray` recently, in #3088.

@telephon mind looking into this? thanks :)

**Comment by telephon** [comment]
>...wait a minute, i could have sworn it was actually loadToFloatArray at some point. when did that change?

Sure it did, in the commit you refer to. `loadToFloatArray` also exists, but requires access to the same file system, while `getToFloatArray` will also allow plotting remote server signals.

I'll take a look.

**Comment by telephon** [comment]
It is an old bug in `getToFloatArray`.

```supercollider
b = Buffer(s, 8727, 2);
b.getToFloatArray(0, 8727, action: { |a| a.postcs });
```

**Comment by telephon** [comment]
Interesting. 

```supercollider
// works fine
(
{
	var src= SinOsc.ar(440), amp= SinOsc.ar(440/20, 0, 2);
	[src, amp, src * amp, (src * amp).clip(-1, 1)]
}.plot(2 * 1633 + 67 / s.sampleRate)
)

/*
[ 'b_getn', 1, 0, 1633 ]
[ 'b_getn', 1, 1633, 1633 ]
[ 'b_getn', 1, 3266, 1633 ]
[ 'b_getn', 1, 4899, 1633 ]
[ 'b_getn', 1, 6532, 1633 ]
[ 'b_getn', 1, 8165, 1633 ]
[ 'b_getn', 1, 9798, 1633 ]
[ 'b_getn', 1, 11431, 1633 ]
[ 'b_getn', 1, 13064, 267 ]
*/
// fails
(
{
	var src= SinOsc.ar(440), amp= SinOsc.ar(440/20, 0, 2);
	[src, amp, src * amp, (src * amp).clip(-1, 1)]
}.plot(2 * 1633 + 69 / s.sampleRate)
)
/*
[ 'b_getn', 1, 0, 1633 ]
[ 'b_getn', 1, 1633, 1633 ]
[ 'b_getn', 1, 3266, 1633 ]
[ 'b_getn', 1, 4899, 1633 ]
[ 'b_getn', 1, 6532, 1633 ]
[ 'b_getn', 1, 8165, 1633 ]
[ 'b_getn', 1, 9798, 1633 ]
[ 'b_getn', 1, 11431, 1633 ]
[ 'b_getn', 1, 13064, 274 ]
FAILURE IN SERVER /b_getn in
*/

```

no time to check further. Maybe someone spots an off by one error somewhere here in `getToFloatArray`?

**Comment by telephon** [comment]
```

(
a = 2 * 1633 + 67 / s.sampleRate;
{
	var src= SinOsc.ar(440), amp= SinOsc.ar(440/20, 0, 2);
	[src, amp]
}.getToFloatArray(a, action: { |x| x.postln })
)


(
a = 2 * 1633 + 69 / s.sampleRate;
{
	var src= SinOsc.ar(440), amp= SinOsc.ar(440/20, 0, 2);
	[src, amp]
}.getToFloatArray(a, action: { |x| x.postln })
)

```

**Comment by brianlheim** [comment]
Closed in #3875.
